code/game/machinery/_machines_base/stock_parts/access_lock.dm

```diff
@@ -25,6 +33,9 @@
 
 /obj/item/stock_parts/access_lock/on_install(obj/machinery/machine)
 	. = ..()
+	if(autoset && istype(machine, /obj/machinery/door))
+		var/obj/machinery/door/D = machine
+		req_access = D.get_auto_access()
```

Comment:
Maybe worth generalizing this to non-doors, where the base implementation is `get_area(machine)?.req_access?.Copy()`.

Reply:
Moved up to machinery level